dynamo/plot/scVectorField.py

```diff
@@ -498,33 +704,46 @@ def cell_wise_vectors(
         sort=sort,
         save_show_or_return="return",
         frontier=frontier,
+        projection=projection,
         **s_kwargs_dict,
         return_all=True,
     )
 
-    if type(axes_list) == list:
-        for i in range(len(axes_list)):
-            axes_list[i].quiver(
-                df.iloc[:, 0],
-                df.iloc[:, 1],
-                df.iloc[:, 2],
-                df.iloc[:, 3],
+    # single axis output
+    if type(axes_list) != list:
+        axes_list = [axes_list]
+    x0, x1 = df.iloc[:, 0], df.iloc[:, 1]
+    v0, v1 = df.iloc[:, 2], df.iloc[:, 3]
+
+    if projection == "3d":
+        x0, x1, x2 = df.iloc[:, 0], df.iloc[:, 1], df.iloc[:, 2]
+        v0, v1, v2 = df.iloc[:, 3], df.iloc[:, 4], df.iloc[:, 5]
+
+    for i in range(len(axes_list)):
+        ax = axes_list[i]
+        if projection == "2d":
+            ax.quiver(
+                x0,
+                x1,
+                v0,
+                v1,
                 color=color_list[i],
                 facecolors=color_list[i],
                 **quiver_kwargs,
             )
-            axes_list[i].set_facecolor(background)
-    else:
-        axes_list.quiver(
-            df.iloc[:, 0],
-            df.iloc[:, 1],
-            df.iloc[:, 2],
-            df.iloc[:, 3],
-            color=color_list,
-            facecolors=color_list,
-            **quiver_kwargs,
-        )
-        axes_list.set_facecolor(background)
+        elif projection == "3d":
+            ax.quiver(
+                x0,
+                x1,
+                x2,
+                v0,
+                v1,
+                v2,
+                # color=color_list[i],
```

Comment:
also, quiver has to be colored the same as the scatters

Reply:
Ok. Extra styling can be added. But we need to confirm if scatters in dynamo cannot work with quiver well due to matplotlib backbone issue. If that is the case, more styling are needed.

Reply:
I was saying revert all changes to streamline

Reply:
you need to take the time and responsibility to fix the bug. This is part of doing science.